conorfirebasewk2/app/src/main/java/com/example/conor_firebase_wk2/NewPost.java

```diff
@@ -37,8 +55,40 @@ public void onClick(View v){
         if (i == R.id.publishPost){
             //code calling method to send the post to the server goes here
 
+            String display = userPost.getText().toString().trim();
+
+            uploadData(display);
+
             //Have the next line as the final call of this button click
             resetText();
         }
     }
+
+    private void uploadData(String display) {
+
+        pd.show();
+
+        String id = UUID.randomUUID().toString();
```

Comment:
this can be changed to the firebase user email...

example of it in View post activity on how to get it from mAuth